procrastinate/app.py

```diff
@@ -4,7 +4,16 @@
 import contextlib
 import functools
 import logging
-from typing import TYPE_CHECKING, Any, Iterable, Iterator
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Iterable,
+    Iterator,
+    TypedDict,
+    Unpack,
```

Comment:
Need to import from typing_extensions, unpack isn't available until 3.11

Reply:
Yep.

I'll give the branch a local run under Python 3.8. I would be surprised if that is the only issue left.